src/analysis/retail/monk/mistweaver/modules/core/HotTrackerMW.tsx

```diff
@@ -38,6 +38,24 @@ class HotTrackerMW extends HotTracker {
     this.risingMistActive = this.owner.selectedCombatant.hasTalent(TALENTS_MONK.RISING_MIST_TALENT);
   }
 
+  dancingMist_RapidDiffusion(hot: Tracker): boolean {
+    return hot.attributions.some(function (attr) {
+      return attr.name === ATTRIBUTION_STRINGS.DANCING_MIST_SOURCES.DM_SOURCE_RD;
+    });
+  }
+
+  dancingMist_HardCast(hot: Tracker): boolean {
+    return hot.attributions.some(function (attr) {
+      return attr.name === ATTRIBUTION_STRINGS.DANCING_MIST_SOURCES.DM_SOURCE_HC;
+    });
+  }
+
+  dancingMist_MistsOfLife(hot: Tracker): boolean {
+    return hot.attributions.some(function (attr) {
+      return attr.name === ATTRIBUTION_STRINGS.DANCING_MIST_SOURCES.DM_SOURCE_MOL;
+    });
+  }
```

Comment:
Not a huge fan of these function names, mixing styles is a bit messy imo

Reply:
renamed